bazel/repository_locations.bzl

```diff
@@ -40,24 +40,37 @@ USE_CATEGORIES = [
 # Components with these use categories are not required to specify the 'cpe' annotation.
 USE_CATEGORIES_WITH_CPE_OPTIONAL = ["build", "test", "other"]
 
+BAZEL_COMPDB_VERSION = "0.4.5"
+BAZEL_GAZELLE_VERSION = "0.19.1"
+BAZEL_TOOLCHAINS_VERSION = "3.4.0"
+
 DEPENDENCY_REPOSITORIES = dict(
     bazel_compdb = dict(
+        project_name = "bazil-compilation-database",
```

Comment:
@moderation @mattklein123 @lizan WDYT about adding this additional metadata to `repository_locations.bzl`? By tracking project name/url/version, we can build out human consumable tables in the docs for the ext deps.

I've updated a few of the deps for now, if we agree I can populate the rest.

Reply:
Rather than add the metadata I wondered if you could extract the value from the `url`. But of course there are inconsistencies between the different `url` sources. Github and Kafka is predictable, the Google repos are a mixed bag and the Python repos a mess. Here is a `/` delimited breakdown https://docs.google.com/spreadsheets/d/1OYS9nU55ibLr4Z-io1esgKPO4FIoWsuOzKG60RxePO8/edit#gid=373925327

Reply:
I think project URL can be more useful to humans if we specify it explicitly, since we can point at non-GitHub pages and also we sometime have the situation where the artifacts we are fetching comes from GCS, even though its canonical home is elsewhere.

Reply:
I think we should specify this manually. We should add linter to make sure that all fields are populated. I think scripts to extract this info will be too cumbersome to maintain.

Reply:
I think this is be a great start point, though long term we might want to split this into a separate file, which may metadata for dependencies even not specified here. `bazel query //external:all --output` gives all the dependencies that bazel loads externally, and that key can be used to join external metadata.

Reply:
Good point, I agree that we're not capturing the complete view here, but let's launch and iterate.